src/api/time/IndependentTimeContext.js

```diff
@@ -20,18 +20,44 @@
  * at runtime from the About dialog for additional information.
  *****************************************************************************/
 
-import TimeContext from "./TimeContext";
+import TimeContext, { TIME_CONTEXT_EVENTS, TIME_CONTEXT_METHODS } from "./TimeContext";
 
 /**
  * The IndependentTimeContext handles getting and setting time of the openmct application in general.
  * Views will use the GlobalTimeContext unless they specify an alternate/independent time context here.
  */
 class IndependentTimeContext extends TimeContext {
-    constructor(globalTimeContext, key) {
+    constructor(globalTimeContext, objectPath) {
         super();
-        this.key = key;
-
+        this.unlisteners = [];
         this.globalTimeContext = globalTimeContext;
+        this.objectPath = objectPath;
+        this.refreshContext = this.refreshContext.bind(this);
+        this.resetContext = this.resetContext.bind(this);
+
+        TIME_CONTEXT_METHODS.forEach((functionName) => {
+            const wrappedFunction = this[functionName].bind(this);
+            this[functionName] = this.handleTimeContextMethods.bind(this, functionName, wrappedFunction);
+        });
+
+        this.refreshContext();
+
+        this.globalTimeContext.on('refreshContext', this.refreshContext);
+    }
+
+    /**
+     * Causes this time context to call methods on another time context (either the global context, or another upstream time context)
+     * This allows views to have their own time context which points to the appropriate upstream context as necessary, achieving nesting.
+     * @param {*} functionName name of the time context class method
+     * @param {*} wrappedFunction the original class method on this (to prevent recursion)
+     * @param {*} args any arguments passed to the called class method
+     */
+    handleTimeContextMethods(functionName, wrappedFunction, ...args) {
+        if (this.upstreamTimeContext) {
```

Comment:
I don't see `upstreamTimeContext` being initialized in the constructor. Although it's not necessary to do so in JS I think it's helpful to developers to have all of the member variables defined in the constructor.

Reply:
Done